openstack/networking/v2/ports/requests.go

```diff
@@ -130,7 +133,36 @@ type CreateOpts struct {
 
 // ToPortCreateMap builds a request body from CreateOpts.
 func (opts CreateOpts) ToPortCreateMap() (map[string]interface{}, error) {
-	return gophercloud.BuildRequestBody(opts, "port")
+	body, err := gophercloud.BuildRequestBody(opts, "port")
+	if err != nil {
+		return nil, err
+	}
+
+	return AddValueSpecs(body)
+}
+
+// AddValueSpecs expands the 'value_specs' object and removes 'value_specs'
+// from the reqeust body. It will return error if the value specs would overwrite
+// an existing field or contains forbidden keys.
+func AddValueSpecs(body map[string]interface{}) (map[string]interface{}, error) {
+	bannedKeys := []string{"shared", "tenant_id"}
```

Comment:
Why only these two?

Reply:
I copied what Heat is doing: https://github.com/openstack/heat/blob/dd7319e373b88812cb18897f742b5196a07227ea/heat/engine/resources/openstack/neutron/neutron.py#L59
Not sure about why exactly these two were banned though

Reply:
Fine by me, the list can evolve later. However I'll like to see a comment explaining that we took it from Heat without knowing much about why these 2 :smile: